backend/scripts/shift-tiers/get-tier-from-liquidity-function.sql

```diff
@@ -0,0 +1,48 @@
+-- Create a more efficient function to get the tier from liquidity
+create
+or replace function get_tier_from_liquidity_efficient (
+  outcome_type text,
+  num_answers integer,
+  liquidity numeric
+) returns text as $$
+DECLARE
+  ante NUMERIC;
+  tier_liquidity NUMERIC;
+  fixed_ante CONSTANT NUMERIC := 1000;
+  multiple_choice_minimum_cost CONSTANT NUMERIC := 1000;
+BEGIN
+  IF outcome_type IN ('POLL', 'BOUNTIED_QUESTION') THEN
+    RETURN NULL;
+  END IF;
+
+  ante := CASE outcome_type
+    WHEN 'BINARY' THEN fixed_ante
+    WHEN 'MULTIPLE_CHOICE' THEN GREATEST(fixed_ante / 10 * COALESCE(num_answers, 0), multiple_choice_minimum_cost)
+    WHEN 'FREE_RESPONSE' THEN fixed_ante / 10
+    WHEN 'PSEUDO_NUMERIC' THEN fixed_ante * 2.5
+    WHEN 'STONK' THEN fixed_ante
+    WHEN 'NUMBER' THEN fixed_ante * 10
+    ELSE fixed_ante
+  END;
+
+  -- Special handling for NUMBER type
+IF outcome_type = 'NUMBER' THEN
+  IF liquidity >= (ante * 10) THEN 
+    RETURN 'crystal';
+  ELSIF liquidity >= ante THEN 
+    RETURN 'premium';
+  ELSIF liquidity >= (ante / 10) THEN 
+    RETURN 'plus';
+  ELSE 
+    RETURN 'play';
+  END IF;
+END IF;
+
+  -- For other types
+  IF liquidity >= ante * 100 THEN RETURN 'crystal';
+  ELSIF liquidity >= ante * 10 THEN RETURN 'premium';
+  ELSIF liquidity >= ante  THEN RETURN 'plus';
+  ELSE RETURN 'play';
+  END IF;
+END;
+$$ language plpgsql immutable;
```

Comment:
This logic looks good, but ofc be sure to test it on dev first. 

Thinking about this a bit more, I don't love how we'll have this logic duplicated in sql and ts, though. If we edit the `FIXED_ANTE` in ts, now we have to edit `fixed_ante` in sql and then also update every contract. That makes experimentation very slow. Have we tried putting `totalLiquidity` into its own native column and then computing the tiers on the fly? That would be much easier than having to precompute all of this.